packages/rum-angular/src/apm.service.ts

```diff
@@ -24,19 +24,19 @@
  */
 
 import { Router, NavigationStart } from '@angular/router'
-import { Injectable } from '@angular/core'
+import { Inject, Injectable } from '@angular/core'
 import { afterFrame } from '@elastic/apm-rum-core'
+import { ApmBase } from '@elastic/apm-rum'
+import { APM } from './apm.module'
 
 @Injectable({
   providedIn: 'root'
 })
 export class ApmService {
-  static apm: any
-
-  constructor(public router: Router) {}
+  constructor(@Inject(APM) public apm: ApmBase, public router: Router) {}
 
   init(config) {
-    const apmInstance = ApmService.apm.init(config)
+    const apmInstance = this.apm.init(config)
```

Comment:
Issues I had regarding Protractor timeouts I traced back to here. When the `init` method is called `zone.js` was triggering change detection on these long running requests. I move this call out of zone and Protractor timeouts went away. Like this:

```
let apmInstance;

this.zone.runOutsideAngular(() => {
  apmInstance = this.apm.init(config);
});
```

And add zone to your constructor dependencies: `private readonly zone: NgZone`

Reply:
Makes sense though on the timeout issue. Seems like the init is dependent on the UI being updated. We dont need the RUM agent to wait for any UI updates. I will move it out.